src/replication.c

```diff
@@ -185,16 +199,20 @@ replicator(void *arg) {
 
 				if(cmd_blocked == false) {
 					TAILQ_INSERT_TAIL(&replica->sendq, rcmd, rsend_cmd_next);
-					if(rcmd->opcode == ZVOL_OPCODE_READ) {
-						read_cmd_sent = true;
-					}
 					pthread_cond_signal(&replica->r_cond);
+					if(cmd_read && !send_multiple_read) {
+						MTX_UNLOCK(&replica->r_mtx);
+						break;
+					}
+				} else {
+					if(cmd_read && (--healthy_untried > 0)) {
```

Comment:
healthy_untried is set to healthy_count, but, its getting decremented if cmd is blocked, which doesn't seem to be correct.
You can just have logic where:
- find whether multiple sends are required or not by looking at healthy_rcount of spec
- if multiple need to be sent, just keep sending to all by adding either to blockedq or sendq
- if multiple need not to be sent, just keep searching for a healthy one, add to its blockedq or sendq and do break
- if no such healthy one found, just iterate once again by setting multiple sends as required